src/aztec/Subsidy.sol

```diff
@@ -287,4 +258,63 @@ contract Subsidy is ISubsidy {
         }
         return withdrawableBalance;
     }
+
+    /**
+     * @notice Gets current accumulated subsidy amount for a given `_bridge` and `_criteria`
+     * @param _bridge Address of the subsidized bridge
+     * @param _criteria A value defining a specific bridge call
+     * @return - Accumulated subsidy amount (would paid out if the bridge called `claimSubsidy(_criteria, ...)`)
+     */
+    function getAccumulatedSubsidyAmount(address _bridge, uint256 _criteria) external view returns (uint256) {
+        Subsidy memory sub = subsidies[_bridge][_criteria];
+        return _computeAccumulatedSubsidyAmount(sub);
+    }
+
+    /**
+     * @notice Sets `Subsidy.gasUsage` value for a given criteria
+     * @dev This function has to be called from the bridge
+     * @param _criteria A value defining a specific bridge call
+     * @param _gasUsage The gas usage of the subsidized action. Used as upper limit for subsidy.
+     * @param _minGasPerMinute Minimum amount of gas per minute that subsidizer has to subsidize
+     */
+    function setGasUsageAndMinGasPerMinute(
+        uint256 _criteria,
+        uint32 _gasUsage,
+        uint32 _minGasPerMinute
+    ) public override(ISubsidy) {
+        // Loading `sub` first in order to not overwrite `sub.available` in case this function was already called
+        // before and subsidy was set. This should not be needed for well written bridges.
+        Subsidy memory sub = subsidies[msg.sender][_criteria];
+
+        sub.gasUsage = _gasUsage;
+        sub.minGasPerMinute = _minGasPerMinute;
+
+        subsidies[msg.sender][_criteria] = sub;
+    }
+
+    /**
+     * @notice Computes accumulated subsidy amount based on Subsidy struct and current block
+     * @param _subsidy Subsidy based on which to compute accumulated subsidy amount
+     * @return - Accumulated subsidy amount
+     */
+    function _computeAccumulatedSubsidyAmount(Subsidy memory _subsidy) private view returns (uint256) {
+        uint256 subsidyAmount;
```

Comment:
This variable seems useless. The `subsidyAmount` is never set, can discard the var, and just return 0 if not already returned inside `dt > 0`

Reply:
Addressed in e183db78c24b4fc5b3795e723f690248b8ed3940